### PR TITLE
Midori Browser Flatpak

### PR DESCRIPTION
**Pull request by Mau-G**
I'd like to submit Midori browser to flathub.

https://github.com/midori-browser/core

### CODE REVIEW

**Comment by Mau-G** [comment]
bot, build org.midori_browser.Midori

File: org.midori_browser.Midori.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{
+  "app-id": "org.midori_browser.Midori",
+  "branch": "stable",
+  "runtime": "org.gnome.Platform",
+  "runtime-version" : "3.28",
```
Comment (gasinvein) [inline org.midori_browser.Midori.json]: Why `3.28`? Doesn't it build with latest SDK, `3.32`?

Comment (Mau-G) [reply org.midori_browser.Midori.json]: it was an oversight.

**Comment by Mau-G** [comment]
bot, build org.midori_browser.Midori

File: org.midori_browser.Midori.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "--system-talk-name=org.freedesktop.GeoClue2",
+    "--talk-name=org.freedesktop.secrets"
+  ],
+  "rename-icon": "midori",
+  "rename-appdata-file": "midori.appdata.xml",
+  "modules": [
+    {
+      "name": "libpeas",
+      "cleanup": [
+        "/bin/*",
```
Comment (TingPing) [inline org.midori_browser.Midori.json]:
```suggestion
        "/bin",
```

File: org.midori_browser.Midori.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        {
+          "type": "git",
+          "url": "https://gitlab.gnome.org/GNOME/libpeas.git",
+          "tag": "libpeas-1.22.0"
+        }
+      ]
+    },
+    {
+      "name": "midori",
+      "buildsystem": "cmake",
```
Comment (TingPing) [inline org.midori_browser.Midori.json]:
```suggestion
      "buildsystem": "cmake-ninja",
```

(may or may not work, but would be a bit faster)

File: README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+    flatpak install flathub org.gnome.Platform//3.28 org.gnome.Sdk//3.28
+
+Build using [flatpak-builder](http://docs.flatpak.org/en/latest/flatpak-builder.html):
+
+    flatpak-builder _build org.midori_browser.Midori.json --force-clean
+
+Test the locally built package:
+
+    flatpak-builder --run _build org.midori_browser.Midori.json midori
```
Comment (TingPing) [inline README.md]: Don't recommend `--run`, it runs in a build environment which has different permissions than the real one. Do `--install --user` and then `flatpak run`.

(Honestly just remove this entire file its mostly useless)

File: org.midori_browser.Midori.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    "--socket=x11",
+    "--socket=wayland",
+    "--socket=pulseaudio",
+    "--filesystem=xdg-download:rw",
+    "--talk-name=org.freedesktop.Notifications",
+    "--system-talk-name=org.freedesktop.GeoClue2",
+    "--talk-name=org.freedesktop.secrets"
+  ],
+  "rename-icon": "midori",
+  "rename-appdata-file": "midori.appdata.xml",
```
Comment (TingPing) [inline org.midori_browser.Midori.json]: Have you spoke with upstream about following the naming recommendations of the specs?

Comment (kalikiana) [reply org.midori_browser.Midori.json]: Upstream is reading the comments now 😛️

I guess we could change that. It seems a bit strange to do with distro packages, but there's no real reason against renaming this file.

Comment (kalikiana) [reply org.midori_browser.Midori.json]: Although renaming the icon would probably break themes in the short term

File: org.midori_browser.Midori.json (hunk 3 above)
Comment (kalikiana) [reply org.midori_browser.Midori.json]: Ninja should work fine actually.

Comment (Mau-G) [reply org.midori_browser.Midori.json]: I'll make the change

File: README.md (hunk 4 above)
Comment (Mau-G) [reply README.md]: I removed the file

File: org.midori_browser.Midori.json (hunk 5 above)
Comment (kalikiana) [reply org.midori_browser.Midori.json]: https://github.com/midori-browser/core/pull/343

**Comment by LeandroStanger** [comment]
bot, build org.midori_browser.Midori

File: org.midori_browser.Midori.json (hunk 5 above)
Comment (Mau-G) [reply org.midori_browser.Midori.json]: does the icon need to follow such name convention of org.midori_browser.Midori.svg? I usually don't see icons named this way

**Comment by Mau-G** [comment]
We changed the metadata license already. As soon as the new version is released it will stop failing.
@kalikiana

File: org.midori_browser.Midori.json (hunk 5 above)
Comment (TingPing) [reply org.midori_browser.Midori.json]: Yes icons must also.

**Comment by nedrichards** [comment]
So is this an app that upstream is interested in maintaining as well? Great news if so!

**Comment by nedrichards** [comment]
bot, build org.midori_browser.Midori

**Comment by AsciiWolf** [comment]
The AppData file seems to be missing: `stat: cannot stat 'builddir/*/share/app-info/xmls/org.midori_browser.Midori.xml.gz': No such file or directory`

Also, there seems to be an issue with the icon name: `WARNING: Icon not matching app id in /srv/buildbot/worker/build-x86_64-5/build/builddir/export/share/applications/org.midori_browser.Midori.desktop: midori`

**Comment by Mau-G** [comment]
@AsciiWolf yes the appdata has been addressed and will be fixed in the next version to be released soon

**Comment by LeandroStanger** [comment]
bot, build org.midori_browser.Midori

**Comment by Mau-G** [comment]
bot, build org.midori_browser.Midori

**Comment by nedrichards** [comment]
Not a blocker, but the app icon is broken in the about dialog.

**Comment by nedrichards** [comment]
View source doesn't work:

> (midori:3): GLib-GIO-CRITICAL **: 07:47:55.656: g_app_info_launch: assertion 'G_IS_APP_INFO (appinfo)' failed

**Comment by kalikiana** [comment]
Thanks for testing! I opened midori-browser/core#349 and midori-browser/core#350 for the above issues. It's perhaps not necessary to do an immediate follow-up release right away (no crashers or big breakage)? Unless something else pops up that affects actual browsing.

**Comment by Mau-G** [comment]
> Thanks for testing! I opened [midori-browser/core#349](https://github.com/midori-browser/core/pull/349) and [midori-browser/core#350](https://github.com/midori-browser/core/pull/350) for the above issues. It's perhaps not necessary to do an immediate follow-up release right away (no crashers or big breakage)? Unless something else pops up that affects actual browsing.

I second this. I would ask if possible to go ahead with the current release while we hunt for other bugs.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.midori_browser.Midori

**Pull request closed without merging**